manager.go

```diff
@@ -1177,35 +1178,38 @@ func (m *jobManager) LaunchJobForUser(req *JobRequest) (string, error) {
 	return "", fmt.Errorf("job started, you will be notified on completion")
 }
 
-func (m *jobManager) clusterNameForUser(user string) (string, error) {
+func (m *jobManager) clusterDetailsForUser(user string) (string, string, error) {
 	if len(user) == 0 {
-		return "", fmt.Errorf("must specify the name of the user who requested this cluster")
+		return "", "", fmt.Errorf("must specify the name of the user who requested this cluster")
 	}
 
 	m.lock.Lock()
 	defer m.lock.Unlock()
 
-	existing, ok := m.requests[user]
+	existing, ok := m.requestsClient().Get(user)
 	if !ok || len(existing.Name) == 0 {
-		return "", fmt.Errorf("no cluster has been requested by you")
+		return "", "", fmt.Errorf("no cluster has been requested by you")
+	}
+	if len(existing.BuildCluster) == 0 {
+		return "", "", fmt.Errorf("unable to determine build cluster for your job")
 	}
-	return existing.Name, nil
+	return existing.Name, existing.BuildCluster, nil
```

Comment:
This is the only place we need to read req.BuildCluster?  You don't need req.BuildCluster because under the lock in this method you can look up whether there is a known job and get buildcluster from it.  Then you only need to have BuildCluster on the job object.